addons/bus/static/tests/bus_monitoring_service.test.js

```diff
@@ -0,0 +1,154 @@
+    await waitForSteps([CONNECTION_STATE.STABLE, "connect"]);
+    lockWebsocketConnect();
+    MockServer.current.env["bus.bus"]._simulateDisconnection(
+        WEBSOCKET_CLOSE_CODES.ABNORMAL_CLOSURE
+    );
+    await waitForSteps([CONNECTION_STATE.LOST]);
+});
+
+test("connection stable after reconnect if previous connection was long enough", async () => {
+    stepConnectionStateChanges();
+    const unlockBusService = lockBusServiceStart();
+    const env = await makeMockEnv();
+    env.services.bus_service.addEventListener("connect", () => asyncStep("connect"));
+    unlockBusService();
+    await env.services.bus_service.start();
+    await waitForSteps([CONNECTION_STATE.STABLE, "connect"]);
+    await advanceTime(STABLE_CONNECTION_TRESHOLD * 2); // Let the connection live for some time in order for it to be considered stable.
+    const unlockWebsocket = lockWebsocketConnect();
+    MockServer.current.env["bus.bus"]._simulateDisconnection(
+        WEBSOCKET_CLOSE_CODES.ABNORMAL_CLOSURE
+    );
+    await waitForSteps([CONNECTION_STATE.LOST]);
+    unlockWebsocket();
+    await runAllTimers();
+    await waitForSteps([CONNECTION_STATE.STABLE]); // Connection lasted more than STABLE_CONNECTION_TRESHOLD, so it's considered stable.
+});
+
+test("connection unstable after reconnect if previous connection was too short", async () => {
+    stepConnectionStateChanges();
+    const unlockBusService = lockBusServiceStart();
+    const env = await makeMockEnv();
+    env.services.bus_service.addEventListener("connect", () => asyncStep("connect"));
+    unlockBusService();
+    await env.services.bus_service.start();
+    await waitForSteps([CONNECTION_STATE.STABLE, "connect"]);
+    const unlockWebsocket = lockWebsocketConnect();
+    MockServer.current.env["bus.bus"]._simulateDisconnection(
+        WEBSOCKET_CLOSE_CODES.ABNORMAL_CLOSURE
+    );
+    await waitForSteps([CONNECTION_STATE.LOST]);
+    unlockWebsocket();
+    await waitForSteps([CONNECTION_STATE.UNSTABLE]); // Connection lasted less than STABLE_CONNECTION_TRESHOLD, so it's considered unstable.
+});
+
+test("unstable connection become stable after some time", async () => {
+    stepConnectionStateChanges();
+    const unlockBusService = lockBusServiceStart();
+    const env = await makeMockEnv();
+    env.services.bus_service.addEventListener("connect", () => asyncStep("connect"));
+    unlockBusService();
+    await env.services.bus_service.start();
+    await waitForSteps([CONNECTION_STATE.STABLE, "connect"]);
+    const unlockWebsocket = lockWebsocketConnect();
+    MockServer.current.env["bus.bus"]._simulateDisconnection(
+        WEBSOCKET_CLOSE_CODES.ABNORMAL_CLOSURE
+    );
+    await waitForSteps([CONNECTION_STATE.LOST]);
+    unlockWebsocket();
+    await waitForSteps([CONNECTION_STATE.UNSTABLE]);
+    await runAllTimers();
+    await waitForSteps([CONNECTION_STATE.STABLE]);
+});
+
+test("brief disconection not considered as lost neither as unstable", async () => {
```

Comment:
`brief disconnect not considered lost nor unstable`

_(typo in `disconection` and rewording to be shorter and easier to read)_